packages/components/src/base_tooltip/index.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useRef, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { createPopper } from '@popperjs/core';
+import { getPopperOptions } from './utils';
+
+const BaseTooltip = (props) => {
+  const {
+    position,
+    bubble,
+    children,
+    tooltipClassName,
+    arrowClassName,
+    contentClassName,
+    ...cleanProps
+  } = props;
+
+  const content = useRef(null);
+  const tooltip = useRef(null);
+  const arrow = useRef(null);
+
+  useEffect(() => {
+    const popper = createPopper(
+      content.current,
+      tooltip.current,
+      getPopperOptions(arrow.current, position),
+    );
+    return () => {
+      if (popper) popper.destroy();
+    };
+  }, [content, tooltip, arrow, position]);
```

Comment:
```suggestion
  }, [position]);
```

refs don't need to be in deps - they will never change (only the object inside the ref)

Reply:
is `content` needed in deps?